### PR TITLE
MULE-10935: Migrate extensions source to use the new Schedulers

### PR DESCRIPTION
**Pull request by elrodro83**
* Migrate remaining cases
* Remove ThreadingProfile form extensions

### CODE REVIEW

File: extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java [hunk 1]
```diff
@@ -220,12 +225,12 @@ public void onStart(SourceCallback<InputStream, ListenerFileAttributes> sourceCa
...
-    executorService =
-        newSingleThreadExecutor(r -> new Thread(r,
-                                                format("%s%s.file.listener", getPrefix(muleContext), flowConstruct.getName())));
+    // TODO MULE-11018 format("%s%s.file.listener", getPrefix(muleContext), flowConstruct.getName())
+    scheduler = schedulerService.ioScheduler();
+
     started = true;
     stopRequested.set(false);
-    executorService.execute(() -> listen(sourceCallback));
+    scheduledListen = scheduler.scheduleWithFixedDelay(() -> listen(sourceCallback), 0, 1, MILLISECONDS);
```
Comment (dfeist) [inline extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java]: Why move from `execute` to `scheduleWithFixedDelay`? Isn't the for loop going to be more efficient than a recurring task, especially if the resolution is so small.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java [hunk 2]
```diff
@@ -40,10 +40,6 @@
   public static final QName MULE_ABSTRACT_POOLING_PROFILE_TYPE =
       new QName(MULE_NAMESPACE, "abstractPoolingProfileType", MULE_PREFIX);
   public static final QName MULE_POOLING_PROFILE_TYPE = new QName(MULE_NAMESPACE, "pooling-profile", MULE_PREFIX);
-  public static final QName MULE_ABSTRACT_THREADING_PROFILE_TYPE =
```
Comment (dfeist) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java]: Check with MG if there is anything else to be cleaned up in modules aside of mule-core related to threading profile.

File: extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java (hunk 1 above)
Comment (marianogonzalez) [inline extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java]: this cannot be a recurring task. This HAS to be a single threaded loop as it was before

File: extensions/sockets/src/main/java/org/mule/extension/socket/api/source/SocketListener.java [hunk 3]
```diff
@@ -74,19 +72,19 @@
...
     workManager = schedulerService.ioScheduler();
 
-    executorService =
-        newSingleThreadExecutor(r -> new Thread(r,
-                                                format("%s%s.socket.listener", getPrefix(muleContext), flowConstruct.getName())));
-
     stopRequested.set(false);
-    executorService.execute(() -> listen(sourceCallback));
+
+    scheduledListen = workManager.scheduleWithFixedDelay(() -> listen(sourceCallback), 0, 1, MILLISECONDS);
```
Comment (marianogonzalez) [inline extensions/sockets/src/main/java/org/mule/extension/socket/api/source/SocketListener.java]: same deal as with file

File: tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergSource.java [hunk 4]
```diff
@@ -33,7 +36,10 @@
   public static final String CORE_POOL_SIZE_ERROR_MESSAGE = "corePoolSize cannot be a negative value";
   public static final String INITIAL_BATCH_NUMBER_ERROR_MESSAGE = "initialBatchNumber cannot be a negative value";
 
-  private ScheduledExecutorService executor;
+  @Inject
+  private SchedulerService schedulerService;
+
+  private Scheduler executor;
```
Comment (dfeist) [inline tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergSource.java]: -> scheduler?

File: extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java [hunk 5]
```diff
@@ -193,20 +194,26 @@
...
   @Connection
   private FileSystem fileSystem;
 
   private FlowConstruct flowConstruct;
   private WatchService watcher;
   private Predicate<FileAttributes> matcher;
   private Set<FileEventType> enabledEventTypes = new HashSet<>();
-  private ExecutorService executorService;
+  private Scheduler scheduler;
+  private Scheduler listenExecutor;
```
Comment (dfeist) [inline extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java]: listenerScheduler or listenerTaskScheduler?

File: extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java [hunk 6]
```diff
@@ -193,20 +194,26 @@
...
   private PrimaryNodeLifecycleNotificationListener clusterListener;
 
   private final Map<WatchKey, Path> keyPaths = new HashMap<>();
   private final AtomicBoolean stopRequested = new AtomicBoolean(false);
   private boolean started = false;
 
+  private Future<?> submittedListen;
```
Comment (dfeist) [inline extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java]: -> listenerTask/listenerFuture etc?

**Pull request merged**